operator-k8s/src/operator.rs

```diff
@@ -98,20 +93,7 @@ impl Operator {
                         .await?;
                     return Ok(());
                 }
-
-                if !self.config.create_crd
-                    && versions.iter().any(|ver| {
-                        matches!(
-                            compare_versions(ver.name.as_str(), current_version.as_ref()),
-                            Ok(Ordering::Greater),
-                        )
-                    })
-                {
-                    panic!(
-                        "The current XlineCluster CRD version {current_version} is not compatible with higher version on k8s. Please use the latest deployment operator or set --create_crd to true."
-                    );
-                }
-
+                assert!(self.config.create_crd || !versions.iter().any(|ver| ver > &current_version), "The current XlineCluster CRD version {current_version} is not compatible with higher version on k8s. Please use the latest xline-operator or set --create_crd to true.");
                 if self.config.create_crd {
                     debug!("create_crd set to true, force patch this CRD");
                     let _crd = crd_api
```

Comment:
if `self.config.create_crd` is `true`, the above assert will failed, so this `if` block will never be executed.

Reply:
if `self.config.create_crd` equals `true`, then the above assertion `assert!(self.config.create_crd || ...)` should always success?

Reply:
Sry, my mistake